### PR TITLE
Add NuRaft library

### PR DESCRIPTION
**Pull request by as51340**
Adds a NuRaft library into Memgraph codebase.

[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [ ] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

File: libs/setup.sh [hunk 1]
```diff
@@ -277,3 +279,10 @@ popd
 #range-v3 release-0.12.0
 range_v3_ref="release-0.12.0"
 repo_clone_try_double "${primary_urls[range-v3]}" "${secondary_urls[range-v3]}" "rangev3" "$range_v3_ref"
+
+# NuRaft
+nuraft_tag="v2.1.0"
+repo_clone_try_double "${primary_urls[nuraft]}" "${secondary_urls[nuraft]}" "nuraft" "$nuraft_tag" true
+pushd nuraft
+./prepare.sh
```
Comment (andrejtonev) [inline libs/setup.sh:287]: How much does this prolong the setup?

File: src/coordination/coordinator_data.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 #include <range/v3/view.hpp>
 #include <shared_mutex>
+#include "libnuraft/nuraft.hxx"
```
Comment (andrejtonev) [inline src/coordination/coordinator_data.cpp:20]: I think our convention is that 3rd party libraries are included as `<lib.hpp>`

**Review by andrejtonev — APPROVED** [review]
Looks good

File: libs/setup.sh (hunk 1 above)
Comment (as51340) [reply libs/setup.sh:287]: Not a lot, this is prepare.sh
```
#!/bin/bash
set -ex

. manifest.sh

PROJECT_DIR=`pwd`

cd ${PROJECT_DIR}

if [ ! -d asio/asio ]; then
    git clone https://github.com/chriskohlhoff/asio.git ./asio
    cd asio
    git checkout ${ASIO_RELEASE}
    cd ..
fi
```

File: src/coordination/coordinator_data.cpp (hunk 2 above)
Comment (as51340) [reply src/coordination/coordinator_data.cpp:20]: Good catch, will change it

**Comment by as51340** [comment]
@kgolubic release note: NuRaft library added for supporting consistent state between coordinators.

**Pull request merged**